src/pyprediktormapclient/auth_client.py

```diff
@@ -1,31 +1,44 @@
-from pydantic import BaseModel, AnyUrl, validate_call, AwareDatetime, field_validator
+from pydantic import BaseModel, AnyUrl, field_validator, ConfigDict
+from typing import Optional
 from pyprediktormapclient.shared import request_from_api
 import datetime
 import requests
 import json
 import re
+from dateutil.parser import parse, ParserError
+from dateutil.tz import tzutc
+
 
 class Ory_Login_Structure(BaseModel):
     method: str
     identifier: str
     password: str
 
+
 class Token(BaseModel):
     session_token: str
-    expires_at: AwareDatetime = None
-    
-    @field_validator('expires_at', mode='before')
+    expires_at: Optional[datetime.datetime] = None
+
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+
+    @field_validator("expires_at", mode="before")
     def remove_nanoseconds(cls, v):
         if v is None:
             return v
-        match = re.match(r"(\d\d\d\d)-(\d\d)-(\d\d)T(\d\d):(\d\d):(\d\d).\d+(\S+)", v)
+        if isinstance(v, datetime.datetime):
+            return v
+        match = re.match(
+            r"(\d\d\d\d)-(\d\d)-(\d\d)T(\d\d):(\d\d):(\d\d).\d+(\S+)", v
+        )
         if match:
-            return datetime.datetime.strptime(match.group(0)[:-4] + match.group(7), "%Y-%m-%dT%H:%M:%S.%f%z")
+            return datetime.datetime.strptime(
+                match.group(0)[:-4] + match.group(7), "%Y-%m-%dT%H:%M:%S.%f%z"
```

Comment:
Same question as above, but for the data format.

Reply:
Deferring this for now.